dataframe-jdbc/src/main/kotlin/org/jetbrains/kotlinx/dataframe/io/db/MariaDb.kt

```diff
@@ -0,0 +1,83 @@
+
+import org.jetbrains.kotlinx.dataframe.io.TableColumnMetadata
+import org.jetbrains.kotlinx.dataframe.schema.ColumnSchema
+import java.sql.ResultSet
+import kotlin.reflect.typeOf
+
+/**
+ * Represents the MariaDb database type.
+ *
+ * This class provides methods to convert data from a ResultSet to the appropriate type for MariaDb,
+ * and to generate the corresponding column schema.
+ */
+public object MariaDb : DbType("mariadb") {
+    override fun convertDataFromResultSet(rs: ResultSet, tableColumn: TableColumnMetadata): Any? {
+        val name = tableColumn.name
+        return when (tableColumn.sqlType) {
+            "BIT" -> rs.getBytes(name)
+            "TINYINT" -> rs.getInt(name)
+            "SMALLINT" -> rs.getInt(name)
+            "MEDIUMINT"-> rs.getInt(name)
+            "MEDIUMINT UNSIGNED" -> rs.getLong(name)
+            "INTEGER", "INT" -> rs.getInt(name)
+            "INTEGER UNSIGNED", "INT UNSIGNED" -> rs.getLong(name)
+            "BIGINT" -> rs.getLong(name)
+            "FLOAT" -> rs.getFloat(name)
+            "DOUBLE" -> rs.getDouble(name)
+            "DECIMAL" -> rs.getBigDecimal(name)
+            "DATE" -> rs.getDate(name).toString()
+            "DATETIME" -> rs.getTimestamp(name).toString()
+            "TIMESTAMP" -> rs.getTimestamp(name).toString()
+            "TIME"-> rs.getTime(name).toString()
+            "YEAR" -> rs.getDate(name).toString()
+            "VARCHAR", "CHAR" -> rs.getString(name)
+            "BINARY" -> rs.getBytes(name)
+            "VARBINARY" -> rs.getBytes(name)
+            "TINYBLOB"-> rs.getBytes(name)
+            "BLOB"-> rs.getBytes(name)
+            "MEDIUMBLOB" -> rs.getBytes(name)
+            "LONGBLOB" -> rs.getBytes(name)
+            "TEXT" -> rs.getString(name)
+            "MEDIUMTEXT" -> rs.getString(name)
+            "LONGTEXT" -> rs.getString(name)
+            "ENUM" -> rs.getString(name)
+            "SET" -> rs.getString(name)
+            else -> throw IllegalArgumentException("Unsupported MariaDB type: ${tableColumn.sqlType}")
+        }
+    }
+
+    override fun toColumnSchema(tableColumnMetadata: TableColumnMetadata): ColumnSchema {
+        return when (tableColumnMetadata.sqlType) {
+            "BIT" -> ColumnSchema.Value(typeOf<ByteArray>())
+            "TINYINT" -> ColumnSchema.Value(typeOf<Int>())
+            "SMALLINT" -> ColumnSchema.Value(typeOf<Int>())
+            "MEDIUMINT"-> ColumnSchema.Value(typeOf<Int>())
+            "MEDIUMINT UNSIGNED" -> ColumnSchema.Value(typeOf<Long>())
+            "INTEGER", "INT" -> ColumnSchema.Value(typeOf<Int>())
+            "INTEGER UNSIGNED", "INT UNSIGNED" -> ColumnSchema.Value(typeOf<Long>())
+            "BIGINT" -> ColumnSchema.Value(typeOf<Long>())
+            "FLOAT" -> ColumnSchema.Value(typeOf<Float>())
+            "DOUBLE" -> ColumnSchema.Value(typeOf<Double>())
+            "DECIMAL" -> ColumnSchema.Value(typeOf<Double>())
+            "DATE" -> ColumnSchema.Value(typeOf<String>())
```

Comment:
I don't fully comprehend the logic but wold this in the end lead to all date types being shown as a `String` instead of the equivalent kotlinx.time classes? (like for instance done in the csv auto inference)